todo/src/main.rs

```diff
@@ -24,33 +30,62 @@ impl Terminal {
         }
     }
 
-    fn ask_new_todo(&mut self) -> Todo {
+    fn ask_new_todo(&mut self) -> Result<Option<Todo>, TerminalError> {
         println!("Write your new todo:");
-        Todo::new(self.input())
+        self.input().map(|user_input| {
+            if user_input.is_empty() {
+                None
+            } else {
+                Some(Todo::new(user_input))
+            }
+        })
     }
 
-    fn show_todo(&mut self, todo: &Todo) {
-        writeln!(self.stdout, "[ ] - {}", todo.message).unwrap()
+    fn show_todo(&mut self, todo: &Todo) -> Result<(), TerminalError> {
+        writeln!(self.stdout, "[ ] - {}", todo.message)
+            .map_err(|error| TerminalError::Stdout(error))
     }
 
     fn user_intention(&mut self) -> bool {
         println!("Do you want to input a new todo? (y/n)");
-        self.input() == "y"
+        let user_input = self.input();
+        if let Ok(input) = user_input {
+            input == "y"
+        } else {
+            false
+        }
```

Comment:
Acho que dá para usar [`matches!`](https://doc.rust-lang.org/std/macro.matches.html) aqui, algo assim:

```suggestion
        matches!(user_input, Ok(input) if input == "y")
```